fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -195,4 +223,243 @@ public void getTriplesOfBinaryDescription() throws Exception {
+        //persist the two operations
+        session1.persist(rdfSourceOperation);
+
+        //perform the create rdf operation
+        session1.persist(rdfSourceOperation2);
+
+        //get triples should now fail because the session is effectively closed.
+        try {
+            session1.commit(CommitOption.UNVERSIONED);
+            fail("session1.commit(...) invocation should fail.");
+        } catch (PersistentStorageException ex) {
+            //attempted rollback should also fail:
+            try {
+                session1.rollback();
+                fail("session1.rollback(...) invocation should fail.");
+            } catch (final PersistentStorageException e) {
+                assertTrue("exception does not contain expected text",
+                        e.getMessage().contains("already committed to the unversioned "));
+            }
+        }
+    }
+
+    @Test
+    public void getTriplesFailsIfCommitHasAlreadyStarted() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn(OCFL_OBJECT_ID);
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        //mock success on commit for the first object session
+        when(mockSessionFactory.create(eq(OCFL_OBJECT_ID), anyString())).thenReturn(objectSession1);
+
+        //pause on commit for 1 second
+        when(objectSession1.commit(any(CommitOption.class))).thenAnswer(new Answer<String>() {
+            @Override
+            public String answer(final InvocationOnMock invocationOnMock) throws Throwable {
+                Thread.sleep(1000);
+                return null;
+            }
+        });
+
+        final PersistentStorageSession session1 = createSession(index, mockSessionFactory);
+        //persist the two operations
+        session1.persist(rdfSourceOperation);
+
+        new Thread(() -> {
+            try {
+                session1.commit(CommitOption.UNVERSIONED);
+            } catch (PersistentStorageException e) {
+            }
+        }).start();
+
+        //get triples should now fail because the commit has started.
+        try {
+            //sleep for a fraction of a second to ensure
+            //commit thread runs first.
+            Thread.sleep(500);
+            session1.getTriples(resourceId, null);
+            fail("session1.getTriples(...) invocation should fail.");
+        } catch (Exception ex) {
```

Comment:
I'd suggest adding an `(expected = PersistentStorageException.class)` to the test method and change the test method to `throws Exception`. At the moment this would pass if any exception were thrown, but we only want some exceptions to pass.

Reply:
Same as similar.

Reply:
In this case how about catching and ignoring PersistentStorageException, but letting other exceptions fail the test?